src/test/extension.test.ts

```diff
@@ -235,14 +291,28 @@ const languageCode = [
             logger.debug("failed to build URL");
             err.printStackTrace();`,
     },
+    {
+        lang: 'C++',
+        code: `   {
+            log_error("Exception occurred!");
+            throw;
+          }`,
+    },
 ]
 
 describe('decorate Editor', () => {
     projects[0].patternProperties.lineMatches = []
     for (const [i, codeExample] of languageCode.entries()) {
-        it('check common pattern matching for ' + languageCode[i].lang, () =>
-            expect(decorateEditor([], codeExample.code)).toEqual([decorationsList[3]])
-        )
+        if (i < 4) {
```

Comment:
This is brittle, the fixtures should directly declare whether they match a supported or an unsupported case, you shouldn't have to rely on the index in the `languageCode` array for that.

Reply:
Do you think separating it in a `supportedLanguage` and `unsupportedLanguage` list to iterate through would make more sense or should I explicitly test each language with its individual mock data not provided from a list?

Reply:
You could do that or add a field `supported: boolean`.
Is there any value in testing unsupported languages?

Reply:
I like to know that it's not matching code that it shouldn't be matching.